sensing/autoware_pointcloud_preprocessor/src/filter.cpp

```diff
@@ -65,7 +65,8 @@
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 autoware::pointcloud_preprocessor::Filter::Filter(
-  const std::string & filter_name, const rclcpp::NodeOptions & options)
+  const std::string & filter_name, const rclcpp::NodeOptions & options,
+  const bool & has_static_tf_only)
```

Comment:
@amadeuszsz Thank you for your fixing. 
- Since the [input and output frame used in filter.cpp are declared as parameters](https://github.com/autowarefoundation/autoware.universe/blob/7f36c52fdedbdc91e598a6c567b21e65d0650888/sensing/pointcloud_preprocessor/src/filter.cpp#L71-L72), IMO `has_static_tf_only` should be declared as parameter and assign value denpend on the `input_frame` and `output_frame` values,
- For example, if the parameters of ground_segmentation (https://github.com/autowarefoundation/autoware.universe/blob/0e1761ae4ba000beec9b23b76a744a407c0934d9/launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py#L268-L269) set to `map` -> `ground_segmentation_node` should be intialized and treated as `has_static_tf_only = false`

Reply:
@badai-nguyen thank you for review!
If at least one filter can find use with & without static TFs, your concerns are correct and we should expose `has_static_tf_only` as parameter. I will do relevant changes 👍🏻

Reply:
@badai-nguyen 
Adressed in https://github.com/autowarefoundation/autoware.universe/pull/8678/commits/f85f3f2afadeb1cccfaa1b2a16cb1de8df11156c
Currently autoware_pointcloud_preprocessor nodes are widely use among standalone Autoware packages and adding `has_static_tf_only` parameter without default value (false) could cause them to stop working. I suggest to address usage of this performance improvement (static TF buffers) in separate PRs to make this transition safer. By default we will use classic TF listener, however if node inherit from Filter class and have same in & out frame, [we still use improvement version](https://github.com/autowarefoundation/autoware.universe/blob/f85f3f2afadeb1cccfaa1b2a16cb1de8df11156c/sensing/autoware_pointcloud_preprocessor/src/filter.cpp#L119-L125).

Reply:
@amadeuszsz I'm sorry for late response.
- I think it is good for now to keep has_static_tf_only with default(false) value. 
- My [concern ](https://github.com/autowarefoundation/autoware.universe/pull/8678#discussion_r1740114981) here is just a posibility of base_link <-> map transformation when parameters changed. 
- NIT: Current default setting, `has_static_tf_only = true` is also acceptable for `scan_ground_segmentation`.